mautic/mautic
# Use the form results tables as report data sources

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR adds an aditional types of report data sources:
![image](https://user-images.githubusercontent.com/96085911/194944116-cab4c2d3-f8e9-4f5b-bb14-d8c6360ae527.png)

This PR also adds a new setting to enable/disable this mode:
![image](https://user-images.githubusercontent.com/96085911/197360235-ec018018-fc72-4cc2-b691-542c0adf18d1.png)



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open the form settings and enable the `Use the form results as report data sources` setting
3. Create a new form and submit it a few times
4. Create a new report choosing a form result as the data source
5. Set the columns and filters 
6. Click the button `Save & close`
7. The report is displayed
8. Open the form settings again and disable the `Use the form results as data sources` setting
9. Try to create a new report
10. There are no form results data sources in the list
10. Open the list of reports
11. Open an existing report that uses the form results as a data source
12. The report is opened successfully, and there is only one form results data sources in the list


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11567"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
